### PR TITLE
Clturney42-signoff-output

### PR DESCRIPTION
**Pull request by xELFHEROx**
requesting signoff for output

### CODE REVIEW

File: Documentation/Signoffs/clturney42-Ouput-signoff.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+
+### 6) Speaker shall have an input power rating minimum of 1 watt and maximum of 2 watts |Origin: Amplifier specifications and power insurance|
+Due to the amplifier being able to supply a continuous power of 1.4 - 2.5W [1], an appropriate speaker must be selected so that the amplifier has enough power to drive it. A good rule of thumb is to have the amplifier used have an output power 1.5-2 times higher than the speakers power rating. With this in mind, the speaker that is used should have a power rating of 2 watts or less. This ensures that the amplifier will be able to consistently drive the speaker with enough power, even if there are slight fluctuations in the system.
+
+## Buildable schematic
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/clturney42-Ouput-signoff.md]: I'm not clear on what the schematic below is and why it is here if it is the internals of a product.

File: Documentation/Signoffs/clturney42-Ouput-signoff.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+For the purpose of the buildable schematic the pmod will be condensed to a single block with just its ports.
+
+<img width="678" alt="Screen Shot 2023-04-16 at 4 56 03 PM" src="https://user-images.githubusercontent.com/123997954/232345262-ff429d1d-7f6d-4138-8613-c1b8cb05caa5.png">
+
+### **Figure 3: pmodAMP2 connected to speaker with appropriate specifications**
+
+The above image is the buildable schematic for the output subsystem. As shown, the input of the subsystem will be connected to the output of the processing and receive an analog voltage signal. Connecting the gain pin to Vcc results in a 6 dB gain and connecting gain to ground results in a 12 dB gain. The shutdown pin will be connected to Vcc, resulting in a logical 1, meaning the shutdown feature will not be active. Vcc will be connected to the incoming power supply which will be supplying a power level between 2.5-5.5 Watts. The stereo 1/8" output J1 will be connected to 3.5mm male plug to bare wire aux cable so that the power and ground cables from the speaker can be connected.
+
+## Analysis
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/clturney42-Ouput-signoff.md]: I don't want to hear about your components. Prove the system will meet the constraints.

**Review by JesseTNRoberts — COMMENTED** [review]
What constraints are derived from the design of the blackfin (I imagine time delay at least)?

**Review by JesseTNRoberts — COMMENTED** [review]
Why are both gain settings connected? The analog in seems to connect to nothing. That's a problem because it can't be built.

The BOM is not sufficient and does not precisely specify components.

Do not explain components. Prove the system meets all your constraints. Long paragraphs are typically inappropriate. This should be numbers, references, and/or simulations with sufficient explanation of what those mean.

**Review by JesseTNRoberts — COMMENTED** [review]
In the analysis, focus on proving the constraints are met. Be very direct and organized about which constraint you are addressing.

**Review by JesseTNRoberts — COMMENTED** [review]
You need to consider how this will mount to the wall. That too must be designed.

How do you know the amp will produce the power sufficient to generate the sound pressure desired from the speaker?

Also, what about time delay?

**Review by JesseTNRoberts — COMMENTED** [review]
Still no design for mounting. How are you claiming to analyze the constraints when the constraints pertain to things that don't exist?

File: Documentation/Signoffs/clturney42-Output-signoff.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+> 7) Speaker shall have an input power rating minimum of 1 watt and maximum of 2 watts 
+  
+  The OWS-5716TA-4C speaker has an input power requirement of 2 watts meaning this constraint is satisfied.
+  
+> 8) Output system shall extend a maximum of 0.5 meters from the wall
+  
+  To make sure that the speaker is not in a place that will impede the movement of those within the classroom, a constraint of 0.5 meters from the wall max has been placed. In the design of the mount subsystem, which will account for speaker mounting, this constraint will be taken into account and met.
+  
+> 9) Output system shall be positioned close enough to wall to reduce delay enough to eliminate echo in sound
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/clturney42-Output-signoff.md]: Why does this need to be based on testing? this should be calculable.

File: Documentation/Signoffs/clturney42-Output-signoff.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  
+> 6) Speaker shall be capable of outputting at a sound pressure efficiency of at least 90 dB
+  
+   According the data sheet of the OWS-5716TA-4C, the speaker has an SPL value of 112.5 dB. This is well over the derived constraint and will provide an excellent sound level that will sufficiently permeate the room.
+
+> 7) Speaker shall have an input power rating minimum of 1 watt and maximum of 2 watts 
+  
+  The OWS-5716TA-4C speaker has an input power requirement of 2 watts meaning this constraint is satisfied.
+  
+> 8) Output system shall extend a maximum of 0.5 meters from the wall
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/clturney42-Output-signoff.md:113]: no analysis

File: Documentation/Signoffs/clturney42-Output-signoff.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  
+  To make sure that the speaker is not in a place that will impede the movement of those within the classroom, a constraint of 0.5 meters from the wall max has been placed. In the design of the mount subsystem, which will account for speaker mounting, this constraint will be taken into account and met.
+  
+> 9) Output system shall be positioned close enough to wall to reduce delay enough to eliminate echo in sound
+  
+  Time delay of speaker sound can be found by dividing distance by speed of sound, which is 343 m/s. Testing will need to be done to see how close the speaker needs to be to the wall so that the delay is small enough to not affect the perceived sound. Factors such as how fast the speed travels through the wall and the time that the signal travels through the circuit will need to be taken into consideration. 
+  
+> 10) Speaker shall be mounted to the wall in such a way that does not impede movement or damage the classroom
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/clturney42-Output-signoff.md]: no analysis

**Review by JesseTNRoberts — COMMENTED** [review]
It seems there should be a time delay aspect to the constraints here.

**Review by JesseTNRoberts — COMMENTED** [review]
Still need a mounting method designed

**Review by JesseTNRoberts — COMMENTED** [review]
How do you know that the speaker will be able to match the amplitude (volume) of the noise?

**Review by JesseTNRoberts — COMMENTED** [review]
comment not addressed.

File: Documentation/Signoffs/clturney42-Output-signoff.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+  
+An ideal noise cancellation device would have no delay however, that is impossible to do when the signal is being processed and traveling through different hardware. With this in mind, the delay of the device needs to be set so that there is as minimal as possible disparity between the inputted signal and outputted noise. This can be verified and changed based on the distance the speaker is positioned from the wall.
+  
+### 10) Speaker shall be mounted, and will not impede movement or damage the classroom
+
+The speaker will be fitted into an enclosure that matches the dimensions of the speaker. This enclosure will then be mounted to the wall in an optimal position for noise cancellation. To fit this constraint, the position in which the speaker is placed will not impede movement or damage the classroom. This can be verified by confirming the placement of the speaker does not block the classroom or damage the walls or windows within the classroom.
+
+### 11) Speaker shall have a sound pressure level and efficiency that is sufficient to permeate the room |Origin: Derived Contraint from room|
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/clturney42-Output-signoff.md]: I think that the question shouldn't be if the system can permeate the room, but how loud of a sound can it cancel.

Comment (xELFHEROx) [reply Documentation/Signoffs/clturney42-Output-signoff.md]: I believe I went over this in the analysis for constraint 11, I have changed the wording of the constraint but the analysis goes over how the SPL of the speaker will maintain a level that is higher than that of construction noises, so it should be able to cancel it.

**Review by JesseTNRoberts — COMMENTED** [review]
Where did the information regarding speaker selection come from?

**Review by JesseTNRoberts — COMMENTED** [review]
It seems to me that you are using terms that you don't understand clearly. 

For you to convince me that this will work, you have to be convinced.

**Review by JesseTNRoberts — APPROVED** [review]

**Pull request merged**